src/server/json_family.cc

```diff
@@ -620,60 +617,102 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
   return vec;
 }
 
-template <typename Op>
+enum ArithmeticOpType { OP_ADD, OP_MULTIPLY };
+
+// Returns true in case of overflow
+bool BinOpApply(double num, bool num_is_double, ArithmeticOpType op, JsonType* val) {
```

Comment:
I would prefer to see result JsonType and overflow flag like out parameter, current implementation looks awkward